### ISSUE TITLE
Features to finalize CMSSW_10_3_0

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#24869]
This issue collects all the needed crucial features needed to build a CMSSW_10_3_0 that can be used for full-workflow tests in view of HI data taking. Part of the input comes from the XC meeting https://indico.cern.ch/event/762341/ see document https://docs.google.com/document/d/1EZYIoJaOkkIQlDgEbRHK0s2bwn5nOctFNX4ou8ZEj88/edit

Known needs so far (2018/10/19 5:10):

- [x] Raw data remapper module  #24819  (merged in 10_4_X) -> #24891 (10_3_X)
- [x] Raw data remapper sequence #24910 (merged in 10_4_X) -> #24921 (10_3_X)
- [x] L1T centrality fix #24837 (merged in 10_3_X/10_4_X)
- [x] SiStrip unpacker #24812 (10_4_X) -> #24845 (10_3_X)
- [x] HI centrality DQM #24835 (merged in 10_4_X) -> #24868 (10_3_X)
- [x] L1T O2O cleaning #24867 (merged in 10_4_X) -> #24875 (10_3_X)
- [x] L1T second HF min-bias feature bit #24847 (merged in 10_4_X) -> #24857 (10_3_X)
- [x] L1T Calo Layer 2 saturation fixes #24883 (merged in 10_4_X) -> #24885 (10_3_X)

**Comment by fabiocos** [comment]
@icali @slava77 @perrotta please list here the possibly missing PRs
@rekovic Is anything more needed from L1T?
@Martin-Grunewald please let me know whether anything is needed by HLT before the 10_3_0 build, or whether more will have to be added afterwards

**Comment by boudoul** [comment]
Adding myself @boudoul as watcher

**Comment by perrotta** [comment]
@fabiocos 
- I confirm that a backport of  #24613 is wanted for 10_3_X, since they want to use it in the online DQM of the last part of this year data taking (i.e. HI). It will have no effect on the HI or standard workflows, though
- #24845 (backport of #24812) is also welcome. Again, in the standard configuration it will have no effect on the HI workflows, and it affects only special (non-production) setups.

**Comment by rekovic** [comment]
@fabiocos 
We would also like a backport of tidy-up of HI L1T O2O https://github.com/cms-sw/cmssw/pull/24867

**Comment by rekovic** [comment]
Also the L1T DQM would need a backport 
https://github.com/cms-sw/cmssw/pull/24868

**Comment by Martin-Grunewald** [comment]
@fabiocos 
At this stage, from the HLT side, we expect several menu updates, esp. of course the HIon menu and related knock-on effects, which may however come only after 10_3_0 is out.

**Comment by mguilbau** [comment]
Hi, I believe we also need https://github.com/cms-sw/cmssw/pull/24883/ in

**Comment by christopheralanwest** [comment]
@fabiocos The inclusion of PR#24857 in 10_3_X has been requested by @mguilbau.

**Comment by mandrenguyen** [comment]
Unfortunately, we will also need #24889 for prompt reco

**Comment by BetterWang** [comment]
Centrality reco 10_4_X (#24892), back port to 10_3_X (#24889).

**Comment by fabiocos** [comment]
@mandrenguyen @mguilbau @icali to be clear, here we should concentrate on the bare minimum for a meaningful test of data taking. Not crucial fixes may come later

**Comment by mandrenguyen** [comment]
@fabiocos Thanks, Fabio.  As you correctly understood, #24889 is not needed for a meaningful test.  Sorry for the noise.

**Comment by fabiocos** [comment]
@perrotta #24613 has been merged, although as it is it looks dummy, so I do not see any reason to rush for a backport (that does not yet exist). Please confirm my understanding, or explain why is it so urgent

**Comment by perrotta** [comment]
@fabiocos : no, it is not urgent.
But it was in the intention of the proponents to have it eventualy in  
CMSSW_10_3_X, to be used for a while in online DQM during the data  
taking.
The backport can wait for the next updates, then

Fabio Cossutti <notifications@github.com> ha scritto:

**Comment by fabiocos** [comment]
@perrotta ok thanks, so it can exit the list of this issue

**Comment by mguilbau** [comment]
Hi, 
https://github.com/cms-sw/cmssw/pull/24883 and https://github.com/cms-sw/cmssw/pull/24857
are fixing the trigger emulator decision for a large fraction of our objects. The agreement between data and emulator online is indeed something we wan't to check during the test. Nevertheless, this PR can be manually merged in DQM online if needed

**Comment by zhenhu** [comment]
Hi @fabiocos , we would also like a backport of the fastSim2017 PU and premix workflows #24829

**Comment by boudoul** [comment]
Dear @icali , please provide the PR for the workflow by tonight as you agreed. Thank you.

**Comment by icali** [comment]
sorry for the delay. The PR is #24910 for 10_4_X

**Comment by boudoul** [comment]
thanks @icali, the schedule is now really critical. Please @icali  follow and address without any delay comments and suggestions from the ongoing review

**Comment by icali** [comment]
backport of #24910 is in PR #24921
